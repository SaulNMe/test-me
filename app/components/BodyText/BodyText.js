import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './BodyTextStyle';

export default class BodyText extends PureComponent {
	render() {
		const textColor = {color: Colors[this.props.color]};
		const weight = {fontWeight: Fonts.weight[this.props.weight] };
		const align = {textAlign: this.props.align };

		return (
			<View>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</View>
		);
	}
}

	BodyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	BodyText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
