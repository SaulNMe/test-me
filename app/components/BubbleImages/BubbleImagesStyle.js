import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	...ApplicationStyles.margins,
	...ApplicationStyles.flexBox,
	bubbleOne: {
		width: 137,
		height: 137,
		borderRadius: 69,
		justifyContent: 'center',
		alignItems: 'center',
		borderWidth: 1
	},
	bubbleTwo: {
		width: 180,
		height: 180,
		borderRadius: 90,
		justifyContent: 'center',
		alignItems: 'center',
		borderWidth: 1
	},
	bubbleThree: {
		width: 107,
		height: 107,
		borderRadius: 54,
		justifyContent: 'center',
		alignItems: 'center',
		borderWidth: 1
	},
	paddingRight: {
		paddingRight:  Metrics.baseMargin
	},
	paddingLeft: {
		paddingLeft:  Metrics.baseMargin
	}
});
