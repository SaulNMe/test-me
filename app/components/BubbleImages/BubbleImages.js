import React, { PureComponent } from 'react';
import { Text, View, Image } from 'react-native';
import PropTypes from 'prop-types';

import styles from './BubbleImagesStyle';

import BodyText from 'qualificame-native/app/components/BodyText';

export default class BubbleImages extends PureComponent {
	render() {
		return (
			<View style={[styles.row, {marginHorizontal: 80, height: 330}]}>
				<View style={styles.justifyContentSpaceBetween}>
					<View>	
						<Image 
							source={require('qualificame-native/assets/image1.png')}
							style={styles.bubbleOne}
							resizeMode='cover'
						/>
					</View>
					<View style={styles.paddingRight}>	
						<BodyText
							text= {this.props.secondText}
							weight= 'light'
							color= {this.props.textColor}
						/>
					</View>
					<View style={{marginLeft: 35}}>	
						<Image 
							source={require('qualificame-native/assets/image3.png')}
							style={styles.bubbleThree}
							resizeMode='cover'
						/>
					</View>
				</View>
				<View style={styles.justifyContentSpaceEvenly}>
					<View style={styles.paddingLeft}>	
						<BodyText
							text= {this.props.firstText}
							weight= 'light'
							color= {this.props.textColor}
						/>
					</View>
					<View>	
						<Image 
							source={require('qualificame-native/assets/image2.png')}
							style={styles.bubbleTwo}
							resizeMode='cover'
						/>
					</View>
					<View style={styles.paddingLeft}>	
						<BodyText
							text= {this.props.thirdText}
							weight= 'light'
							color= {this.props.textColor}
						/>
					</View>
				</View>
			</View>
		);
	}
}
	BubbleImages.propTypes = {
		firstText: PropTypes.string,
		secondText: PropTypes.string,
		thirdText: PropTypes.string,
		textColor: PropTypes.string
	}

	BubbleImages.defaultProps = {
		firstText: '1 Lorem ipsum dolor.',
		secondText: '2 Lorem ipsum dolor sit.',
		thirdText: '3 Lorem ipsum dolor sit.',
		textColor: 'red'
	}
