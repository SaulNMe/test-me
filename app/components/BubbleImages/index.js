import BubbleImages from './BubbleImages';

export default BubbleImages;
