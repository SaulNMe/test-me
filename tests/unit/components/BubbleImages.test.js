import React from 'react';
import { shallow } from 'enzyme';

import BubbleImages from 'qualificame-native/app/components/BubbleImages';

import BodyText from 'qualificame-native/app/components/BodyText';
import { Text, Image } from 'react-native';

function createTestProps (props) {
	return {
		// common props
		firstText: 'Text No 1',
		secondText: 'Text No 2',
		thirdText: 'Text No 3',
		textColor: 'red',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<BubbleImages {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders ad expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays text', () => {
		expect(wrapper.find(BodyText).get(0).props.text).toBeTruthy();
		expect(wrapper.find(BodyText).get(1).props.text).toBeTruthy();
		expect(wrapper.find(BodyText).get(2).props.text).toBeTruthy();
	});

	it('displays images', () => {
		expect(wrapper.find(Image).get(0)).toBeTruthy();
		expect(wrapper.find(Image).get(1)).toBeTruthy();
		expect(wrapper.find(Image).get(2)).toBeTruthy();
	});
})