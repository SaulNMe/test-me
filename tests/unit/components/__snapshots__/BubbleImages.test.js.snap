// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders ad expected 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "height": 330,
        "marginHorizontal": 80,
      },
    ]
  }
>
  <Component
    style={
      Object {
        "justifyContent": "space-between",
      }
    }
  >
    <Component>
      <Component
        resizeMode="cover"
        source={1}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 90,
            "borderWidth": 1,
            "height": 137,
            "justifyContent": "center",
            "width": 137,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "paddingRight": 16,
        }
      }
    >
      <BodyText
        align=""
        color="red"
        text="Text No 2"
        weight="light"
      />
    </Component>
    <Component
      style={
        Object {
          "marginLeft": 35,
        }
      }
    >
      <Component
        resizeMode="cover"
        source={1}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 90,
            "borderWidth": 1,
            "height": 107,
            "justifyContent": "center",
            "width": 107,
          }
        }
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "space-evenly",
      }
    }
  >
    <Component
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    >
      <BodyText
        align=""
        color="red"
        text="Text No 1"
        weight="light"
      />
    </Component>
    <Component>
      <Component
        resizeMode="cover"
        source={1}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 90,
            "borderWidth": 1,
            "height": 180,
            "justifyContent": "center",
            "width": 180,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    >
      <BodyText
        align=""
        color="red"
        text="Text No 3"
        weight="light"
      />
    </Component>
  </Component>
</Component>
`;
